worlds/Files.py

```diff
@@ -115,8 +117,14 @@ def write_contents(self, opened_zipfile: zipfile.ZipFile) -> None:
             manifest_str = json.dumps(manifest)
         except Exception as e:
             raise Exception(f"Manifest {manifest} did not convert to json.") from e
-        else:
-            opened_zipfile.writestr("archipelago.json", manifest_str)
+
+        manifest_location = "archipelago.json"
+        if self.manifest_write_subdirectory is not None:
+            if len(Path(self.manifest_write_subdirectory).parts) != 1:
```

Comment:
I personally would've gone for
`if "/" in self.manifest_write_subdirectory or "\\" in self.manifest_write_subdirectory:`

I think this also misses the case where `manifest_write_subdirectory` is `""` ?

Reply:
> I personally would've gone for `if "/" in self.manifest_write_subdirectory or "\\" in self.manifest_write_subdirectory:`

Oh, I was actually trying to make it so "dir/" is a valid value. It would become "dir//archipelago.json" but that would still resolve correctly